rust/connlib/tunnel/src/client.rs

```diff
@@ -181,6 +180,7 @@ where
     #[tracing::instrument(level = "trace", skip(self))]
     pub fn set_interface(&mut self, config: InterfaceConfig) -> connlib_shared::Result<()> {
         self.role_state.interface_config = Some(config);
+        self.update_dns_mapping();
```

Comment:
In this case we always update the interface because we've had a network change or something and _must_ rebuild the interface?

Reply:
yeah, basically, we could check for changes on the config though when #4322 is updated